library/ssl_cli.c

```diff
@@ -2183,8 +2183,8 @@ static int ssl_write_encrypted_pms( mbedtls_ssl_context *ssl,
 
     if( ssl->session_negotiate->peer_cert == NULL )
     {
-        MBEDTLS_SSL_DEBUG_MSG( 2, ( "certificate required" ) );
-        return( MBEDTLS_ERR_SSL_UNEXPECTED_MESSAGE );
+        /* Should never happen */
+        return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
```

Comment:
Minor: returning internal error is usually preceded by `MBEDTLS_SSL_DEBUG_MSG( 1, ( "should never happen" ) );`: if it ever happens, we want to easily know on which line (with `DEBUG_C` enabled).

Reply:
Ok, I'll add it in.

Reply:
Done.